File: app/src/main/java/kashyap/in/androidbestpractices/base/listeners/NetworkBroadcast.java
package kashyap.in.androidbestpractices.base.listeners;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class NetworkBroadcast extends BroadcastReceiver {

    NetworkChangeListener networkChangeListener;

    @Override
    public void onReceive(Context context, Intent intent) {
        if (networkChangeListener != null) {
            networkChangeListener.onNetworkChanged();
        }
    }

    public void setSharedListener(NetworkChangeListener networkChangeListener) {
        this.networkChangeListener = networkChangeListener;
    }

    public interface NetworkChangeListener {
        void onNetworkChanged();
    }
}

File: app/src/main/java/kashyap/in/androidbestpractices/base/listeners/NetworkBroadcast.kt
package kashyap.`in`.androidbestpractices.base.listeners

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class NetworkBroadcast : BroadcastReceiver() {
    var networkChangeListener: NetworkChangeListener? = null
    override fun onReceive(context: Context, intent: Intent) {
        if (networkChangeListener != null) {
            networkChangeListener!!.onNetworkChanged()
        }
    }

    fun setSharedListener(networkChangeListener: NetworkChangeListener?) {
        this.networkChangeListener = networkChangeListener
    }

    interface NetworkChangeListener {
        fun onNetworkChanged()
    }
}